frontend-minesweeper/src/App.js

```diff
@@ -0,0 +1,168 @@
+import './App.css';
+import winner from './winner.gif';
+import explode from './explode.gif';
+
+function App() {
+  const [apiResponse, setApiResponse] = useState(null);
+  const [tileData, setTileData] = useState([]);
+  const [level, setLevel] = useState(0);
+  const [gameId, setGameId] = useState(null);
+  const [gameWon, setGameWon] = useState(false);
+  const [gameOver, setGameOver] = useState(false);
+  const [time, setTime] = useState(0);
+  const [timerOn, setTimerOn] = useState(false);
+  const [showPopup, setShowPopup] = useState(false);
+  const intervalRef = useRef(null);
+
+  const API_HOST = 'http://localhost:5008';
+  const gameSettings = [
+    { rows: 9, columns: 9, mines: 5 },
+    { rows: 18, columns: 18, mines: 15 },
+    { rows: 60, columns: 60, mines: 99 },
+  ];
+
+  useEffect(() => {
+    if (timerOn && !intervalRef.current) {
+      intervalRef.current = setInterval(() => {
+        setTime(time => time + 1);
+      }, 1000);
+    }
+    return () => {
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+        intervalRef.current = null;
+      }
+    };
+  }, [timerOn]);
+
+  useEffect(() => {
+    createNewGame();
+  }, [level]);
+
+  useEffect(() => {
+    checkVictory();
+    if (apiResponse) {
+      if (apiResponse.gameWon) {
+        console.log('Du vant!');
+        setGameOver(false);
+        setTimerOn(false);
+        setShowPopup(true);
+        setTimeout(() => setShowPopup(false), 1000);
+      }
+      if (apiResponse.mineExploded) {
+        console.log('Spillet er over');
+        setGameOver(true);
+        setTimerOn(false);
+        setShowPopup(true);
+        setTimeout(() => setShowPopup(false), 1000); 
+        const updatedTiles = tileData.map(tile => ({
+          ...tile,
+          isRevealed: tile.mine ? true : tile.isRevealed
+        }));
+        setTileData(updatedTiles);
+      }
+    }
+  }, [apiResponse]);
+
+  const createNewGame = () => {
+    if (intervalRef.current) clearInterval(intervalRef.current);
+    intervalRef.current = null;
+    setApiResponse(null);
+    setGameWon(false);
+    setGameOver(false);
+    setShowPopup(false);
+    setTileData([]);
+    setTime(0);
+    setTimerOn(false);
+    const data = { level, userName: "sigmaronny" };
+
+    axios.post(`${API_HOST}/games/`, data)
+      .then(response => {
+        setApiResponse(response.data);
+        setGameId(response.data.gameId);
+        if (response.data.board && response.data.board.tiles) {
+          setTileData(response.data.board.tiles);
+        }
+        setTimerOn(true);
+      })
+      .catch(error => console.error("Feil ved oppretting av spill:", error));
+  };
+
+  const handleTileAction = (row, column, move) => {
+    if (gameWon || gameOver) { return; }
+    const flaggedTiles = tileData.filter(tile => tile.isFlagged).length;
+    if (move === 1 && flaggedTiles >= gameSettings[level].mines) {
+      return;
+    }
+    const MoveData = { row, column, move };
```

Comment:
Dette er ikke en komponent, så den [bør](https://dev.to/kristiyan_velkov/react-js-naming-convention-lcg) hete `moveData`.